lib/Monad_WP/NonDetMonadVCG.thy

```diff
@@ -797,6 +800,18 @@ lemma empty_fail_return [simp, wp]:
   "empty_fail (return x)"
   by (simp add: empty_fail_def return_def)
 
+lemma empty_fail_bindE:
+  "\<lbrakk> empty_fail f; \<And>rv. empty_fail (g rv) \<rbrakk>
+       \<Longrightarrow> empty_fail (f >>=E g)"
```

Comment:
indent

Reply:
Hm. I've explicitly tried not to do that, but the imports into this file are pretty small. On the other hand the entire file needs massive cleanup and should eventually be much smaller, so I'm not sure it makes sense to mix that into this one.